libsolidity/interface/StandardCompiler.cpp

```diff
@@ -614,8 +614,135 @@ std::variant<OptimiserSettings, Json::Value> parseOptimizerSettings(Json::Value
 	return { std::move(settings) };
 }
 
+/// Only parses the input settings and fills the InputsAndSettings with it.
+/// If this fails, it does return a JSON error response, otherwise nullopt on success.
+optional<Json::Value> parseInputSettings(Json::Value _settings, StandardCompiler::InputsAndSettings& _output)
```

Comment:
Please try to avoid output parameters at all costs. Really. At all costs. There is no need for this in C++17.

Reply:
This is a rather big function - maybe it's better to have this as a static member in the class?